tests/integrationv2/test_renegotiate_apache.py

```diff
@@ -0,0 +1,83 @@
+import pytest
+import tempfile
+
+from configuration import ALL_TEST_CURVES
+from common import ProviderOptions
+from fixtures import managed_process
+from providers import Provider, S2N
+from utils import invalid_test_parameters, get_parameter_name
+from constants import TEST_CERT_DIRECTORY
+
+from test_renegotiate import TEST_PROTCOLS, S2N_RENEG_OPTION, S2N_RENEG_ACCEPT
+
+APACHE_SERVER_IP = "127.0.0.1"
+APACHE_SERVER_PORT = 7777
+
+APACHE_SERVER_CERT = TEST_CERT_DIRECTORY + "apache_server_cert.pem"
+APACHE_CLIENT_CERT = TEST_CERT_DIRECTORY + "apache_client_cert.pem"
+APACHE_CLIENT_KEY = TEST_CERT_DIRECTORY + "apache_client_key.pem"
+
+
+def create_get_request(route):
+    return f"GET {route} HTTP/1.1\r\nHost: localhost\r\n\r\n"
+
+
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("curve", ALL_TEST_CURVES, ids=get_parameter_name)
+@pytest.mark.parametrize("protocol", TEST_PROTCOLS, ids=get_parameter_name)
+def test_change_cipher_suite_endpoint(managed_process, curve, protocol):
+    options = ProviderOptions(
+        mode=Provider.ClientMode,
+        host=APACHE_SERVER_IP,
+        port=APACHE_SERVER_PORT,
+        curve=curve,
+        protocol=protocol,
+        trust_store=APACHE_SERVER_CERT,
+    )
+
+    options.extra_flags = [S2N_RENEG_OPTION, S2N_RENEG_ACCEPT]
```

Comment:
Have you manually tested that this times out / repros the failures we're targeting without these arguments?

Do you think that's worth writing a failure integration test?

Reply:
I did test them manually, but it definitely makes sense to add failing tests. It takes some time for apache to give up on the renegotiation and return 403, but running this for 3 versions on two endpoints isn't too slow. 91c393cac0bec9ab5f8a17cde675043bef4a60e2